api/views.py

```diff
@@ -127,3 +132,79 @@ def post(self, request, *args, **kwargs):
         data = serializer.data
 
         return Response(data, status=status.HTTP_201_CREATED)
+
+    @staticmethod
+    def _get_account_from_card_id(card_uuid) -> SociBankAccount:
+        soci_bank_account = get_object_or_404(queryset=SociBankAccount.objects.all(), card_uuid=card_uuid)
+
+        return soci_bank_account
+
+
+class SensorMeasurementView(generics.CreateAPIView, generics.ListAPIView):
+    serializer_class = SensorMeasurementSerializer
+
+    def get_permissions(self):
+        if self.request.method == "POST":
+            return [SensorTokenPermission()]
+
+        return []
+
+    def get_queryset(self):
+        # We don't really bother to validate this input. A bad input will just
+        # return 0 results.
+        type = self.request.query_params.get('type', MEASUREMENT_TYPE_TEMPERATURE)
+        return SensorMeasurement.objects.filter(
+            type=type,
+            created_at__gte=timezone.now() - timezone.timedelta(days=1)
+        )
+
+    @swagger_auto_schema(
+        request_body=serializer_class(many=True),
+        operation_summary="Create measurements",
+        responses={
+            "201": serializer_class(many=True),
+            "400": ": Illegal input.",
+            "403": ": You are not authorized to create measurements."
+        },
+    )
+    def post(self, request: Request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data, many=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        # Try to clear out old entries.
+        try:
+            SensorMeasurementView._remove_old_measurement_instances()
+        except DatabaseError:
+            # Should we do something?
+            pass
+
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    @swagger_auto_schema(
+        operation_summary="Get measurement data for the last 24 hours.",
+        manual_parameters=[Parameter(
+            name='type',
+            in_=IN_QUERY,
+            default="temperature",
+            description=f"The type of measurements.",
+            type=TYPE_STRING,
+            enum=[x[0] for x in MEASUREMENT_TYPE_CHOICES],
+            required=False
+        )],
+        responses={
+            "200": serializer_class(many=True),
```

Comment:
I believe that when this is the only response, it will be automatically generated by drf-yasg, Cf. line 42

Reply:
I see. I think I'll let it stay here though — unless you consider that a problem. Having things like this explicitly makes it easier for DRF/Swagger-beginners like myself to see what is supposed to be output.